src/main/java/touro/chess/BishopPiece.java

```diff
@@ -18,36 +18,36 @@ public BishopPiece(Location location, PieceColor color)
 //        int col = this.getLocation().getColumn();
 
         //Possible moves to the upper left
-        for(int row = this.getLocation().getRow(), col = this.getLocation().getColumn();
-            col >= 1; row--, col--)
+        for(int row = (this.getLocation().getRow()) - 1, col = (this.getLocation().getColumn()) - 1;
```

Comment:
Is this better than creating a variable outside the loop to hold the current row and column? (and then using that to reset the variables used in your loops later on) 
for ex: `currentRow = this.getLocation.getRow()`and then using that in your loops
int row = currentRow -1 
Don't know which way is more efficient.

Reply:
I like this way I think it eliminates unnecessary code

Reply:
Yes. You should avoid repeatedly calling methods that will return the same result each time.